lib/framework/string_ext.h

```diff
@@ -46,8 +46,8 @@
 #	endif
 #	if defined(DEBUG)
 #		define strdup(s) \
-	strdup2(s,__FILE__,__LINE__)
-static inline char *strdup2(const char *s, char *fileName, int line)
+	strdup2(static_cast<const char*>(s),__LINE__)
```

Comment:
when changing the C++ version of the standard, I discovered that it didn't compile (because type is const char[] vs const char* or something), I reverted back to C++14 but I think this is a safe change